pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

```diff
@@ -0,0 +1,1827 @@
+			fields: fields{
+				subjectAllowList: map[string]bool{
+					"spirex1@example.com": true,
+					"spirex2@example.com": true,
+					"spirex3@example.com": true,
+					"spirex4@example.com": true,
+					"spirex5@example.com": true,
+				},
+			},
+			args: args{
+				subject: "spirex4@example.com",
+			},
+			want: map[string]bool{
+				"spirex1@example.com": true,
+				"spirex2@example.com": true,
+				"spirex3@example.com": true,
+				"spirex4@example.com": true,
+				"spirex5@example.com": true,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				subjectAllowList: tt.fields.subjectAllowList,
+			}
+			sigstore.AddAllowedSubject(tt.args.subject)
+			if !reflect.DeepEqual(sigstore.subjectAllowList, tt.want) {
+				t.Errorf("sigstore.subjectAllowList = %v, want %v", sigstore.subjectAllowList, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_ClearAllowedSubjects(t *testing.T) {
+	type fields struct {
+		subjectAllowList map[string]bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   map[string]bool
+	}{
+
+		{
+			name: "clear existing map",
+			fields: fields{
+				subjectAllowList: map[string]bool{
+					"spirex1@example.com": true,
+					"spirex2@example.com": true,
+					"spirex3@example.com": true,
+					"spirex4@example.com": true,
+					"spirex5@example.com": true,
+				},
+			},
+			want: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				subjectAllowList: tt.fields.subjectAllowList,
+			}
+			sigstore.ClearAllowedSubjects()
+			if !reflect.DeepEqual(sigstore.subjectAllowList, tt.want) {
+				t.Errorf("sigstore.subjectAllowList = %v, want %v", sigstore.subjectAllowList, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_EnableAllowSubjectList(t *testing.T) {
+	type fields struct {
+		allowListEnabled bool
+	}
+	type args struct {
+		flag bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   bool
+	}{
+		{
+			name: "disabling subject allow list",
+			fields: fields{
+				allowListEnabled: true,
+			},
+			args: args{
+				flag: false,
+			},
+			want: false,
+		},
+		{
+			name: "enabling subject allow list",
+			fields: fields{
+				allowListEnabled: false,
+			},
+			args: args{
+				flag: true,
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				allowListEnabled: tt.fields.allowListEnabled,
+			}
+			sigstore.EnableAllowSubjectList(tt.args.flag)
+			if sigstore.allowListEnabled != tt.want {
+				t.Errorf("sigstore.allowListEnabled = %v, want %v", sigstore.allowListEnabled, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_SelectorValuesFromSignature(t *testing.T) {
+	type fields struct {
+		allowListEnabled bool
+		subjectAllowList map[string]bool
+	}
+	type args struct {
+		signature oci.Signature
+	}
+	tests := []struct {
```

Comment:
this function is creating several logs in case of failures but this test is not capturing them.
I think we need to capture errors to know what it is happening here